listenbrainz/webserver/views/art_api.py

```diff
@@ -773,44 +339,25 @@ def cover_art_yim(user_name, year: int = 2024):
     branding = _parse_bool_arg("branding", True)
     legacy = _parse_bool_arg("legacy", False)
 
-    stats = db_yim.get(user["id"], year, legacy=legacy)
+    if legacy:
+        if year < 2021 or year > 2024:
+            raise APIBadRequest("Legacy Year In Music images are only available for years 2021, 2022, 2023 and 2024")
+        return cover_art_yim_legacy(user=user, year=year, image=image, branding=branding)
+
+    stats = db_yim.get(user["id"], year, legacy=False)
     if stats is None:
         raise APIBadRequest(f"Year In Music {year} report for user {user_name} not found")
-
-    season = request.args.get("season") or 'spring'
-    match season:
-        case "spring":
-            background_color = "#EDF3E4"
-            accent_color = "#2B9F7A"
-        case "summer":
-            background_color = "#DBE8DF"
-            accent_color = "#3C8C54"
-        case "autumn":
-            background_color = "#F1E8E1"
-            accent_color = "#CB3146"
-        case "winter":
-            background_color = "#DFE5EB"
-            accent_color = "#5B52AC"
-
-    # todo: use legacy query param here when new YIM images are ready
-    yim24 = None
-    if year == 2024:
-        yim24 = {
-            "season": season,
-            "background_color": background_color,
-            "accent_color": accent_color,
-            "image_folder": f'{current_app.config["SERVER_ROOT_URL"]}/static/img/year-in-music-24/{season}',
-        }
-
-    match image:
-        case "overview": svg = _cover_art_yim_overview(user_name, stats, year, yim24)
-        case "stats": svg = _cover_art_yim_stats(user_name, stats, year, yim24)
-        case "albums": svg = _cover_art_yim_albums(user_name, stats, year, yim24)
-        case "tracks": svg = _cover_art_yim_tracks(user_name, stats, year, yim24)
-        case "artists": svg = _cover_art_yim_artists(user_name, stats, year, yim24)
-        case "discovery-playlist": svg = _cover_art_yim_playlist(user_name, stats, "playlist-top-discoveries-for-year", year, branding, yim24)
-        case "missed-playlist": svg = _cover_art_yim_playlist(user_name, stats, "playlist-top-missed-recordings-for-year", year, branding, yim24)
-        case other: raise APIBadRequest(f"Invalid image type {other}. Image type should be one of (stats, artists, albums, tracks, discovery-playlist, missed-playlist)")
+    
+    svg = None
+    # match image:
```

Comment:
For now I've commented these out, as I have removed these _cover_art_yim_... methods from this file, and I will recreate them based on what's in my yim-25 page.
In any case, there should be no path hitting non-legacy YIM art API, and it will just return a 204 + empty content in any case.